### DISCUSSION TITLE
Failed to resolve module specifier swup

### DISCUSSION THREAD

**Discussion by MPFabularisLab** [swup/swup#647] [Help] [answered]
Hello everyone, i need help

I want to add swup transition page in my Astro websites, I installed the pacakges, add a app.js file in public folder where I inizialize Swup.
After it I added script module in my shared Layout file but the console show this error : 

"caught TypeError: Failed to resolve module specifier "swup". Relative references must start with either "/", "./", or "../"."

/public/app.js
`import Swup from 'swup'; 
const swup = new Swup();`

src/layouts/Layout.astro
`<script type="module" src="/app.js"></script>`

Console.error:
caught TypeError: Failed to resolve module specifier "swup". Relative references must start with either "/", "./", or "../".

**Comment by daun** [comment, marked as answer]
Sounds like either `swup` isn't installed correctly in your `package.json` or Astro isn't set up to bundle npm dependencies. Hard to say from here. Have you posted this in an Astro forum or support channel? They will be much better suited to help you out as this is an Astro issue and not a swup problem.

**Reply by MPFabularisLab** [reply]
Yes it was problem how I integrate it with Astro, now it works!
Only issue now it's with theme and plugins, they didn't work. When I try to import it like I do with Swup (and it works)
`import SwupSlideTheme from '@swup/slide-theme';`
my editor suggest me this:

Could not find a declaration file for module '@swup/slide-theme'. 'c:/Users/Utente/Documents/Astro/fabularis-astro/node_modules/@swup/slide-theme/lib/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/swup__slide-theme` if it exists or add a new declaration (.d.ts) file containing `declare module '@swup/slide-theme';`
  
package.json
  `"dependencies": {
"@swup/slide-theme": "^1.0.5",
}`

**Reply by daun** [reply]
That just means there's no TypeScript types as the plugins are all in vanilla JS. It should still be usable but won't offer any type hints in your IDE.

**Reply by MPFabularisLab** [reply]
Then Slide theme and Scroll plugin don't work for another reason.
Could be possible SSR don't work fine with it?

**Reply by daun** [reply]
Not sure. But generally, swup has no concept of SSR so I'd make sure to only load it in client context.
